src/ensembl/src/content/app/entity-viewer/gene-view/components/proteins-list/proteins-list-item/ProteinsListItem.tsx

```diff
@@ -15,90 +15,90 @@
 import { toggleExpandedProtein } from 'src/content/app/entity-viewer/state/gene-view/proteins/geneViewProteinsSlice';
 import { getExpandedTranscriptIds } from 'src/content/app/entity-viewer/state/gene-view/proteins/geneViewProteinsSelectors';
 
-import { RootState } from 'src/store';
+import ProteinsListItemInfo from '../proteins-list-item-info/ProteinsListItemInfo';
+import { TranscriptQualityLabel } from 'src/content/app/entity-viewer/shared/components/default-transcript-label/TranscriptQualityLabel';
 import { Transcript } from 'src/content/app/entity-viewer/types/transcript';
 import { View } from 'src/content/app/entity-viewer/state/gene-view/view/geneViewViewSlice';
 
 import transcriptsListStyles from 'src/content/app/entity-viewer/gene-view/components/default-transcripts-list/DefaultTranscriptsList.scss';
 import styles from './ProteinsListItem.scss';
 
-const UNIPROT_SOURCE = 'Uniprot';
+const SWISSPROT_SOURCE = 'Uniprot/SWISSPROT';
 
 type Props = {
   transcript: Transcript;
   isDefault?: boolean;
   trackLength: number;
-  expandedTranscriptIds: string[];
-  toggleExpandedProtein: (id: string) => void;
 };
 
 const ProteinsListItem = (props: Props) => {
-  const { transcript, trackLength } = props;
-
+  const { isDefault, transcript, trackLength } = props;
+  const expandedTranscriptIds = useSelector(getExpandedTranscriptIds);
   const dispatch = useDispatch();
+
   const params: { [key: string]: string } = useParams();
   const { genomeId, entityId } = params;
   const { search } = useLocation();
   const proteinIdToFocus = new URLSearchParams(search).get('protein_id');
 
   const { product } = transcript.product_generating_contexts[0];
+
   const toggleListItemInfo = () => {
     if (proteinIdToFocus) {
       const url = urlFor.entityViewer({
         genomeId,
         entityId,
         view: View.PROTEIN
       });
+
       dispatch(replace(url));
     }
-    props.toggleExpandedProtein(product.stable_id);
+
+    dispatch(toggleExpandedProtein(product.stable_id));
   };
 
   const midStyles = classNames(transcriptsListStyles.middle, styles.middle);
-
   const itemRef = useRef<HTMLDivElement | null>(null);
+
   useEffect(() => {
     if (product.stable_id === proteinIdToFocus) {
       setTimeout(() => {
         itemRef.current?.scrollIntoView({
           behavior: 'smooth'
         });
       }, 100);
-      if (!props.expandedTranscriptIds.includes(proteinIdToFocus)) {
-        props.toggleExpandedProtein(product.stable_id);
+
+      if (!expandedTranscriptIds.includes(proteinIdToFocus)) {
+        dispatch(toggleExpandedProtein(product.stable_id));
       }
     }
   }, [proteinIdToFocus]);
 
   const getProteinDescription = () => {
-    const uniprotReference = product.external_references.find((reference) =>
-      reference.source.id.includes(UNIPROT_SOURCE)
+    const swissprotReference = product.external_references.find((reference) =>
+      reference.source.id.includes(SWISSPROT_SOURCE)
```

Comment:
why `.includes` and not `===`?

![image](https://user-images.githubusercontent.com/6834224/104311140-b021d180-54cc-11eb-9c20-41f678117594.png)

Reply:
It's because the `UNIPROT_SOURCE` constant had only `Uniprot` before. But now we can switch to using `===` as you've suggested.

Reply:
✅